### PR TITLE
Client

### PR DESCRIPTION
**Pull request by MarrMika**

### CODE REVIEW

File: src/main/java/com/pnu/ordermanagementapp/adapter/DbAdapter.java [hunk 1]
```diff
@@ -4,14 +4,12 @@
 
 public interface DbAdapter<T> { // added only to meet task requirements
 
-    T findById(String id);
```
Comment (Igor-Polatajko) [inline src/main/java/com/pnu/ordermanagementapp/adapter/DbAdapter.java]: implement this method, we need it

File: src/main/java/com/pnu/ordermanagementapp/client/ClientDbAdapter.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @Override
+    public void delete(Long id) {
+        Client client = findClientByIdOrThrowException(id);
+        repository.delete(client);
+    }
+
+    private Client findClientByIdOrThrowException(Long id){
+        return repository.findById(id)
```
Comment (Igor-Polatajko) [inline src/main/java/com/pnu/ordermanagementapp/client/ClientDbAdapter.java:48]: will it bring user to some custom page or the user will get full stack trace in browser?
I'd suggest just skip deleting, if there is no client, with provided id for now, later we will handle edge cases somehow

File: src/main/java/com/pnu/ordermanagementapp/model/Client.java [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
 import java.util.List;
 
 @Data
-@Builder
```
Comment (Igor-Polatajko) [inline src/main/java/com/pnu/ordermanagementapp/model/Client.java:10]: Why removed?

File: src/main/resources/templates/show_clients.ftl [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                        Client email:
+                        <div class="px-2 rounded bg-white mx-3">
+                            <strong>${client.email}</strong>
+                        </div>
+                    </div>
+                </div>
+            </div>
+                <div class="col-md-2 d-flex justify-content-center">
+                    <div class="align-self-center">
+                            <button class="btn btn-danger" href="new_order">Create order</button>
```
Comment (Igor-Polatajko) [inline src/main/resources/templates/show_clients.ftl]: change the color of button to blue ("btn-primary")

File: src/main/resources/templates/show_clients.ftl [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                            <form method="POST" action="/clients/delete/${client.id}">
```
Comment (Igor-Polatajko) [inline src/main/resources/templates/show_clients.ftl]: Add button "orders" (just simple styled as a button html link) to link to "/orders/client/{id}"

File: src/main/java/com/pnu/ordermanagementapp/adapter/DbAdapter.java (hunk 1 above)
Comment (MarrMika) [reply src/main/java/com/pnu/ordermanagementapp/adapter/DbAdapter.java]: You are right

File: src/main/java/com/pnu/ordermanagementapp/client/ClientDbAdapter.java (hunk 2 above)
Comment (MarrMika) [reply src/main/java/com/pnu/ordermanagementapp/client/ClientDbAdapter.java:48]: I've implemented an error page!

File: src/main/resources/templates/show_clients.ftl (hunk 4 above)
Comment (MarrMika) [reply src/main/resources/templates/show_clients.ftl]: Ok

File: src/main/resources/templates/show_clients.ftl (hunk 5 above)
Comment (MarrMika) [reply src/main/resources/templates/show_clients.ftl]: Ok

File: src/main/java/com/pnu/ordermanagementapp/model/Client.java (hunk 3 above)
Comment (MarrMika) [reply src/main/java/com/pnu/ordermanagementapp/model/Client.java:10]: It's useless

**Review by Igor-Polatajko — APPROVED** [review]

File: src/main/java/com/pnu/ordermanagementapp/model/Client.java (hunk 3 above)
Comment (Igor-Polatajko) [reply src/main/java/com/pnu/ordermanagementapp/model/Client.java:10]: I don't see a big problem, we'll need it anyway later. But let it be.

**Pull request merged**